src/logger/connectivityTask.c

```diff
@@ -521,149 +521,109 @@ void cellular_buffering_task(void *params)
 
         bool logging_enabled = false;
 
-        uint32_t re_open_buffer_file_timeout = 0;
-        uint32_t last_open_buffer_attempt = getCurrentTicks();
-        uint32_t buffer_file_open_retries = 0;
-
         while (1) {
                 cellular_state.should_stream = logging_enabled ||
                                      logger_config->ConnectivityConfigs.telemetryConfig.backgroundStreaming ||
                                      connParams->always_streaming;
 
-                while (1) {
-                        if (!cellular_state.buffer_file_open && isTimeoutMs(last_open_buffer_attempt, re_open_buffer_file_timeout)) {
-                                last_open_buffer_attempt = getCurrentTicks();
-                                cellular_reset_buffer_offset_map();
-                                cellular_state.read_index = 0;
-                                fs_lock();
-                                bool fs_good = sdcard_fs_mounted();
-                                if (!fs_good) {
-                                        FRESULT initfs_rc = InitFS();
-                                        fs_good = (initfs_rc == FR_OK);
-                                        if (FR_OK != initfs_rc) {
-                                                if (!buffer_file_open_retries)
-                                                        pr_info_int_msg(_LOG_PFX "Error Initializing filesystem: ", initfs_rc);
+                if (!cellular_state.buffer_file_open ) {
+                        cellular_reset_buffer_offset_map();
+                        cellular_state.read_index = 0;
+                        bool sd_write_validated = sdcard_write_ready( false ); // (false ==  honor retry delay)
```

Comment:
@brentpicasso I have changed the old functionality here, but it mimics the old functionaly closely as this was the source of the new shared code.